mono/mini/mini-x86.c

```diff
@@ -1925,30 +1925,43 @@ if (ins->inst_true_bb->native_offset) { \
 	x86_fnstsw (code); \
 } while (0); 
 
-
 static guint8*
-emit_call (MonoCompile *cfg, guint8 *code, guint32 patch_type, gconstpointer data)
+x86_align_for_patch (MonoCompile *cfg, guint8 *code, guint32 patch_type, gconstpointer data, int align_offset)
 {
-	gboolean needs_paddings = TRUE;
```

Comment:
These changes don't seem to be neccessary.

Reply:
x86_align_and_patch is reused.
x86_align_for_patch granted only has one use and could be put back inline.
Would that suffice?
I do also like "short circuit with return" instead of updating local booleans but that is style.

This line:
`	guint32 pad_size = (guint32)(code + align_offset - cfg->native_code) & 0x3;
`
confuses me some.
Why the subtract of native_code?
Is it not aligned, but later one there will be some aligned base?
That would explain it.